components/shared_code/src/shared/model/measurement.py

```diff
@@ -294,3 +294,11 @@ def summarize(self) -> dict[str, str | dict[str, str | None]]:
             "start": self["start"],
             "end": self["end"],
         }
+
+    def summarize_latest(self) -> Self:
+        """Return a summary of this measurement, given that it is the latest measurement."""
+        if parameter_hash := self.get("source_parameter_hash"):  # pragma: no feature-test-cover
+            if parameter_hash != self.metric.source_parameter_hash():
+                self["outdated"] = True
```

Comment:
Does it not make more sense to make `outdated` a property, and unset `source_parameter_hash` in the setter method of the prop? This in order to avoid implementing deletion of the `source_parameter_hash` in various places later

Reply:
I don't think that's needed. There's only one location where the hash is deleted and that is when a new one is added.